pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1031,6 +1032,68 @@ func (m *kubeGenericRuntimeManager) GetPodStatus(uid kubetypes.UID, name, namesp
 	}, nil
 }
 
+// FetchPodStatus retrieves the status of the pod, including the
+// information of all containers in the pod that are visible in Runtime.
+func (m *kubeGenericRuntimeManager) FetchPodStatus(pod *kubecontainer.Pod) (*kubecontainer.PodStatus, error) {
```

Comment:
IIUC the only big difference between `FetchPodStatus` and `GetPodStatus` is call to `podSandboxIDs, err := m.getSandboxIDByPodUID(uid, nil)`, so the assumptions that `pod *kubecontainer.Pod` will already have updated sandbox statues.
1. Does it always correct?
2. If pod have updated sandbox statues, does the same true for container statuses(maybe we can also avoid call to `m.getContainerStatuses`)

Reply:
Thanks.  I think this change is based on the assumption that there is no container's change between this PLEG relist() and updateCache(), it's a tradeoff between performance and accuracy. As between this updateCache() and podworker getting podstatus from podCache, It may still have container's change happened, and the podCache maybe outdated.